bot/exts/utils/snekbox.py

```diff
@@ -79,6 +86,109 @@ def print_last_line():
 REDO_EMOJI = '\U0001f501'  # :repeat:
 REDO_TIMEOUT = 30
 
+PythonVersion = Literal["3.10", "3.11"]
+
+
+@dataclass
+class EvalJob:
+    """Job to be evaluated by snekbox."""
+
+    args: list[str]
+    files: list[FileAttachment] = field(default_factory=list)
+    name: str = "eval"
+    version: PythonVersion = "3.11"
+
+    @classmethod
+    def from_code(cls, code: str, path: str = "main.py") -> EvalJob:
+        """Create an EvalJob from a code string."""
+        return cls(
+            args=[path],
+            files=[FileAttachment(path, code.encode())],
+        )
+
+    def as_version(self, version: PythonVersion) -> EvalJob:
+        """Return a copy of the job with a different Python version."""
+        return EvalJob(
+            args=self.args,
+            files=self.files,
+            name=self.name,
+            version=version,
+        )
+
+    def to_dict(self) -> dict[str, list[str | dict[str, str]]]:
+        """Convert the job to a dict."""
+        return {
+            "args": self.args,
+            "files": [file.to_dict() for file in self.files],
+        }
+
+
+@dataclass(frozen=True)
+class EvalResult:
```

Comment:
This and `EvalJob` could be moved to a new module for the sake of readability of the main module.

Reply:
Moved to utils/snekbox/_eval.py in [727a146](https://github.com/python-discord/bot/pull/2326/commits/727a146f2de0f37c43d6939dc4368ef780373cd4)